src/idl_parser.cpp

```diff
@@ -2755,6 +2755,42 @@ std::vector<IncludedFile> Parser::GetIncludedFiles() const {
   return {it->second.cbegin(), it->second.cend()};
 }
 
+bool Parser::HasCircularStructDependency() {
+  std::function<bool(StructDef*)> visit =
+      [&](StructDef* struct_def) {
+        // Only consider fixed structs
+        if (!struct_def->fixed) {
+          return false;
+        }
+
```

Comment:
Don't you also want to just return false if already `Checked` here? If not, this will doubly check every reference to a `struct` and could have explosive cost.